org.eclipse.xtext/src/org/eclipse/xtext/formatting2/internal/HiddenRegionFormattingMerger.java

```diff
@@ -28,13 +30,16 @@ public HiddenRegionFormattingMerger(AbstractFormatter2 formatter) {
 
 	@Override
 	public IHiddenRegionFormatting merge(List<? extends IHiddenRegionFormatting> conflicting) {
+		// If there are only 2 conflicts,
+		// usages of this method expect the second value to be updated to the merge result
+		// TODO: Fix those usages so they no longer expect this method to edit its input
 		if (conflicting.size() == 2) {
-			// TODO: don't do this
 			conflicting.get(1).mergeValuesFrom(conflicting.get(0));
 			return conflicting.get(1);
 		}
 		IHiddenRegionFormatting result = formatter.createHiddenRegionFormatting();
-		for (IHiddenRegionFormatting conflict : conflicting)
+		// Reversed so the merging order is consistent with the above if
```

Comment:
There seems to be something missing from the comment.

Reply:
Ah got it now, `with the above if condition` or `with the special case above` might be a better wording.

Reply:
I fixed it